lib/omnibus/config.rb

```diff
@@ -498,12 +498,17 @@ def reset!
 
     # The number of worker threads for make. If this is not set
     # explicitly in config, it will attempt to determine via Ohai in
-    # the builder, and failing that will default to 3
+    # the builder, and failing that will default to 3. This will
+    # also default to 3 if the count is 0 (like in a Solaris zone)
     #
     # @return [Integer]
     default(:workers) do
-      if Ohai['cpu'] && Ohai['cpu']['total']
-        Ohai['cpu']['total'].to_i + 1
+      count = 0
+      if Ohai['cpu'] && Ohai['cpu'].key?(:total)
```

Comment:
if its a Mash this should probably be 'total' since that's consistent with the rest of the style here, and (our) Mashes convert to strings under the covers anyway.